docs/build/testing-framework.md

```diff
@@ -0,0 +1,108 @@
+# Testing Framework Documentation
+
+## Overview
+
+This document outlines the testing framework used for Subquery projects. The testing framework provides an easy way to test the behavior of mapping handlers and validate the data being indexed.
+
+## Usage
+
+To use the testing framework, you need to:
+
+1. Create test files with the naming convention *.test.ts and place them in the src/tests folder. Each test file should contain test cases for specific mapping handlers.
+2. Run the testing service using the command: `subql-node test`.
+
+## Writing Test Cases
+
+Test cases should be written using the `subqlTest()` function with the following parameters:
+
+- **name**: A string containing the name of the test.
+- **blockHeight**: The height of the block used in the test.
+- **dependentEntities**: An array of entities that are required for the test to run.
+- **expectedEntities**: An array of entities that are expected after the mapping handler has run.
+- **handler**: The name of the mapping handler function to test.
+
+### Example test case
+
+```javascript
+
+// Example handleEvent function
+/*
+export async function handleEvent(event: SubstrateEvent): Promise<void> {
+  const {
+    event: {
+      data: [account, balance],
+    },
+  } = event;
+  //Retrieve the record by its ID
+  const record = await StarterEntity.get(
+    event.block.block.header.hash.toString()
+  );
+  record.field2 = account.toString();
+  //Big integer type Balance of a transfer event
+  record.field3 = (balance as Balance).toBigInt();
+  await record.save();
+}
+
+*/
+
+subqlTest(
+    "handleEvent test", // test name
+    1000003, // block height to process
+    [
+      StarterEntity.create({
+        id: '0x0ca3c88eaa25af380f243273e83ba2e161b207d935357af063e2bd5b8a2e9c40', // Replace this with the actual block hash for block 103
+        field1: 1000003,
+        field2: null,
+        field3: null,
+        field4: null,
+        field5: null,
+      }),
+    ], // dependent entities
+    [
+      StarterEntity.create({
+        id: '0x0ca3c88eaa25af380f243273e83ba2e161b207d935357af063e2bd5b8a2e9c40', // Replace this with the actual block hash for block 103
+        field1: 1000003,
+        field2: '23M5ttkmR6KcoTAAE6gcmibnKFtVaTP5yxnY8HF1BmrJ2A1i', 
+        field3: BigInt(2303515439), 
+        field4: null,
+        field5: null,
+      }),
+    ], // expected entities
+    'handleEvent', //handler name
+  );
+```
+
+## Running Tests
+
+To run the tests, execute the subql-node test command. The testing service will discover and import test files, run the tests, and log the results.
+
+Example output:
+
+Passing test:
+
+![image](https://user-images.githubusercontent.com/40036742/232681495-969e28eb-6ff4-442f-b006-2e92548e9ef9.png)
```

Comment:
I think these images need to be added to the repo. We cant guarantee they remain on github servers